File: Semestrul 4/MPP/Proiecte/serverC#clientJava/c#/IProjectServices.cs
using modelnet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace servicesnet;

public interface IProjectServices
{
    Voluntar login(String username, String password, IProjectObserver client);

    //void logout(IProjectObserver client) throws Exception;
    Donatie saveDonatie(Donatie donatie);

    List<Caz> findAllCaz();

    List<CazDTO> createCazDTOList();

    List<Donator> searchByName(String partialName);

    Donator saveDonator(Donator donator);

    Donator findByFullName(String numeDonator);

    int getSumaDonatiiPentruCaz(int cazId);


    void logout(String username, IProjectObserver client);
    IEnumerable<Voluntar> findAllVoluntari();
}

File: Semestrul 4/MPP/Proiecte/serverC#clientJava/java/Services/src/main/java/services/IProjectServices.java
package services;

import model.*;

import java.util.List;

public interface IProjectServices {
    Voluntar login(String username, String password, IProjectObserver client) throws Exception;

    //void logout(IProjectObserver client) throws Exception;
    Donatie saveDonatie(Donatie donatie) throws Exception;

    List<Caz> findAllCaz() throws Exception;

    List<CazDTO> createCazDTOList() throws Exception;

    List<Donator> searchByName(String partialName) throws Exception;

    Donator saveDonator(Donator donator) throws Exception;

    Donator findByFullName(String numeDonator) throws Exception;

    int getSumaDonatiiPentruCaz(int cazId) throws Exception;


    void logout(String username, IProjectObserver client) throws Exception;

    //List<Voluntar> findAllVoluntari() throws Exception;
}


